File: src/Cxx/Utilities/FunctionParser.cxx
#include <vtkSmartPointer.h>
#include <vtkFunctionParser.h>


int main(int, char *[])
{
  vtkSmartPointer<vtkFunctionParser> functionParser = 
      vtkSmartPointer<vtkFunctionParser>::New();
  functionParser->SetFunction("a+b");
  
  functionParser->SetScalarVariableValue( "a", 2);
  functionParser->SetScalarVariableValue( "b", 3);
    
  double result = functionParser->GetScalarResult();
  
  std::cout << "result: " << result << std::endl;
  
  return EXIT_SUCCESS;
}

File: src/Java/Utilities/FunctionParser.java
import vtk.vtkNativeLibrary;
import vtk.vtkFunctionParser;

public class FunctionParser 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
   vtkFunctionParser functionParser = new vtkFunctionParser();
   functionParser.SetFunction("a+b");
   functionParser.SetScalarVariableValue( "a", 2);
   functionParser.SetScalarVariableValue( "b", 3);
   
   double result = functionParser.GetScalarResult();
   
   System.out.println("Result:" +" " +result);
  }
}
